src/txHashBuilder.c

```diff
@@ -862,9 +887,88 @@ static void txHashBuilder_assertCanLeaveWithdrawals(tx_hash_builder_t* builder)
 	ASSERT(builder->remainingWithdrawals == 0);
 }
 
-void txHashBuilder_addAuxData(tx_hash_builder_t* builder, const uint8_t* auxDataHashBuffer, size_t auxDataHashBufferSize)
+void txHashBuilder_enterMint(tx_hash_builder_t* builder)
 {
 	txHashBuilder_assertCanLeaveWithdrawals(builder);
+	{
+		// Enter mint
+		BUILDER_APPEND_CBOR(CBOR_TYPE_UNSIGNED, TX_BODY_KEY_MINT);
+	}
+	builder->state = TX_HASH_BUILDER_IN_MINT;
+}
+
+void txHashBuilder_addMint_topLevelData(
+        tx_hash_builder_t* builder, uint16_t numAssetGroups
+)
+{
+	_TRACE("state = %u", builder->state);
+
+	ASSERT(builder->state == TX_HASH_BUILDER_IN_MINT);
+
+	builder->outputData.remainingAssetGroups = numAssetGroups;
+	// Map(numAssetGroups)[
+	//   { * policy_id => { * asset_name => uint } }
+	// ]
+	BUILDER_APPEND_CBOR(CBOR_TYPE_MAP, numAssetGroups);
+	if (0 != numAssetGroups) {
+		builder->state = TX_HASH_BUILDER_IN_MINT_ASSET_GROUP;
+	} else {
+		// TODO is this an error case?
+	}
+}
+
+void txHashBuilder_addMint_tokenGroup(
+        tx_hash_builder_t* builder,
+        const uint8_t* policyIdBuffer, size_t policyIdSize,
+        uint16_t numTokens
+)
+{
+	addTokenGroup(builder, policyIdBuffer, policyIdSize, numTokens,
+	              TX_HASH_BUILDER_IN_MINT_ASSET_GROUP, TX_HASH_BUILDER_IN_MINT_TOKEN);
+}
+
+void txHashBuilder_addMint_token(
+        tx_hash_builder_t* builder,
+        const uint8_t* assetNameBuffer, size_t assetNameSize,
+        int64_t amount
+)
+{
+	//  TODO int64 -> uint64 conversion should work fine ?
+	// 	  BUILDER_APPEND_CBOR
+	addToken(builder, assetNameBuffer, assetNameSize, amount,
+	         TX_HASH_BUILDER_IN_MINT_TOKEN,
+	         TX_HASH_BUILDER_IN_MINT_ASSET_GROUP,
+	         TX_HASH_BUILDER_IN_MINT);
+}
+
+static void txHashBuilder_assertCanLeaveMint(tx_hash_builder_t* builder)
+{
+	_TRACE("state = %u, remainingMintAssetGroups = %u, remainingMintTokens = %u",
+	       builder->state, builder->outputData.remainingAssetGroups, builder->outputData.remainingTokens);
+
+	switch (builder->state) {
+	case TX_HASH_BUILDER_IN_MINT:
+		break;
+
+	case TX_HASH_BUILDER_IN_WITHDRAWALS:
+	case TX_HASH_BUILDER_IN_CERTIFICATES:
+	case TX_HASH_BUILDER_IN_TTL:
+	case TX_HASH_BUILDER_IN_FEE:
+		txHashBuilder_assertCanLeaveWithdrawals(builder);
+		break;
+
+	default:
+		ASSERT(false);
+	}
+
+	ASSERT(builder->outputData.remainingAssetGroups == 0);
+	ASSERT(builder->outputData.remainingTokens == 0);
+}
+
+
+void txHashBuilder_addAuxData(tx_hash_builder_t* builder, const uint8_t* auxDataHashBuffer, size_t auxDataHashBufferSize)
+{
+	txHashBuilder_assertCanLeaveMint(builder);
```

Comment:
Mint is the last item in transaction_body in 
https://github.com/input-output-hk/cardano-ledger-specs/blob/master/shelley-ma/shelley-ma-test/cddl-files/shelley-ma.cddl